server/player/player.go

```diff
@@ -122,20 +123,29 @@ func New(name string, skin skin.Skin, pos mgl64.Vec3) *Player {
 	p.velocity.Store(mgl64.Vec3{})
 	p.immunity.Store(time.Now())
 	p.breakingPos.Store(cube.Pos{})
+	if data != nil {
+		p.load(*data)
+		p.loadInventory(data.Inventory)
+	}
 	return p
 }
 
 // NewWithSession returns a new player for a network session, so that the network session can control the
 // player.
 // A set of additional fields must be provided to initialise the player with the client's data, such as the
-// name and the skin of the player.
-func NewWithSession(name, xuid string, uuid uuid.UUID, skin skin.Skin, s *session.Session, pos mgl64.Vec3) *Player {
-	p := New(name, skin, pos)
+// name and the skin of the player. You can either pass on player data you want to load or
+// you can leave the data as nil to use default data.
+func NewWithSession(name, xuid string, uuid uuid.UUID, skin skin.Skin, s *session.Session, pos mgl64.Vec3, data *Data) *Player {
```

Comment:
Can remove the pointer here now as well because NewWithSession will generally always have Data.